src/JWT/Payload.php

```
@@ -0,0 +1,230 @@
+     * @return string
+     */
+    public function jsonSerialize()
+    {
+        return $this->payload;
+    }
+
+    /**
+     * Get the payload as JSON
+     *
+     * @return string
+     */
+    public function __toString()
+    {
+        return $this->toJson();
+    }
+
+    /**
+     * Dynamically check a claim exist
+     *
+     * @param string $claim
+     * @return bool
+     */
+    public function __isset(string $claim)
+    {
+        return $this->offsetExists($claim);
+    }
+
+    /**
+     * Dynamically get a claim
+     *
+     * @param string $claim
+     * @return mixed
+     */
+    public function __get(string $claim)
+    {
+        return $this->offsetGet($claim);
+    }
+
+    /**
+     * Dynamically set a claim
+     *
+     * @param string $claim
+     * @param mixed $value
+     * @return void
+     */
+    public function __set(string $claim, $value)
+    {
+        $this->offsetSet($claim, $value);
+    }
+
+    /**
+     * Dynamically unset a claim
+     *
+     * @param string $claim
+     * @return void
+     */
+    public function __unset(string $claim)
+    {
+        $this->offsetUnset($claim);
+    }
+
+    /**
+     * Determine if the offset exists
+     *
+     * @param string $key
+     * @return void
+     */
+    public function offsetExists($key)
+    {
+        return isset($this->payload[$key]);
+    }
+
+    /**
+     * Set the offset
+     *
+     * @param string $key
+     * @return mixed
+     */
+    public function offsetGet($key)
+    {
+        return $this->payload[$key] ?? null;
+    }
+
+    /**
+     * Set the offset
+     *
+     * @param string $key
+     * @param string $value
+     * @return void
+     */
+    public function offsetSet($key, $value)
+    {
+        $this->payload[$key] = $value;
+    }
+
+    /**
+     * Unset the offset
+     *
+     * @param string $key
+     * @return void
+     */
+    public function offsetUnset($key)
+    {
+        unset($this->payload[$key]);
+    }
+}
```

An exception should be throw if trying to set/unset properties on the payload - it should be readonly